File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/CreateOntologyProjectDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * 
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CreateOntologyProjectDetails.Builder.class)

public final class CreateOntologyProjectDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"workspaceId", "key", "displayName", "description", "namespace", "creator", "ontologyVersion", "baseUri", "defaultLanguage", "workspaceBasePath", "sourceType", "gitRepositoryKey", "gitBranchName", "gitFolderPath"})
    public CreateOntologyProjectDetails(String workspaceId, String key, String displayName, String description, String namespace, String creator, String ontologyVersion, String baseUri, String defaultLanguage, String workspaceBasePath, OntologyProjectSourceType sourceType, String gitRepositoryKey, String gitBranchName, String gitFolderPath) {
        super();
        this.workspaceId = workspaceId;
        this.key = key;
        this.displayName = displayName;
        this.description = description;
        this.namespace = namespace;
        this.creator = creator;
        this.ontologyVersion = ontologyVersion;
        this.baseUri = baseUri;
        this.defaultLanguage = defaultLanguage;
        this.workspaceBasePath = workspaceBasePath;
        this.sourceType = sourceType;
        this.gitRepositoryKey = gitRepositoryKey;
        this.gitBranchName = gitBranchName;
        this.gitFolderPath = gitFolderPath;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            
@com.fasterxml.jackson.annotation.JsonProperty("workspaceId")
private String workspaceId;



public Builder workspaceId(String workspaceId) {
    this.workspaceId = workspaceId;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;



public Builder key(String key) {
    this.key = key;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;



public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;



public Builder description(String description) {
    this.description = description;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("namespace")
private String namespace;



public Builder namespace(String namespace) {
    this.namespace = namespace;
    return this;
}
            /**
     * Creator name written into the generated project.yaml metadata file.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("creator")
private String creator;

        /**
         * Creator name written into the generated project.yaml metadata file.
         * @param creator the value to set
         * @return this builder
         **/
        

public Builder creator(String creator) {
    this.creator = creator;
    return this;
}
            /**
     * Initial semantic ontology version written into project.yaml.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("ontologyVersion")
private String ontologyVersion;

        /**
         * Initial semantic ontology version written into project.yaml.
         * @param ontologyVersion the value to set
         * @return this builder
         **/
        

public Builder ontologyVersion(String ontologyVersion) {
    this.ontologyVersion = ontologyVersion;
    return this;
}
            /**
     * Base URI written as baseURI in project.yaml.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("baseUri")
private String baseUri;

        /**
         * Base URI written as baseURI in project.yaml.
         * @param baseUri the value to set
         * @return this builder
         **/
        

public Builder baseUri(String baseUri) {
    this.baseUri = baseUri;
    return this;
}
            /**
     * Default language tag written into project.yaml.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("defaultLanguage")
private String defaultLanguage;

        /**
         * Default language tag written into project.yaml.
         * @param defaultLanguage the value to set
         * @return this builder
         **/
        

public Builder defaultLanguage(String defaultLanguage) {
    this.defaultLanguage = defaultLanguage;
    return this;
}
            /**
     * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("workspaceBasePath")
private String workspaceBasePath;

        /**
         * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
         * @param workspaceBasePath the value to set
         * @return this builder
         **/
        

public Builder workspaceBasePath(String workspaceBasePath) {
    this.workspaceBasePath = workspaceBasePath;
    return this;
}
            /**
     * Project content source. Defaults to VOLUME when omitted.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sourceType")
private OntologyProjectSourceType sourceType;

        /**
         * Project content source. Defaults to VOLUME when omitted.
         * @param sourceType the value to set
         * @return this builder
         **/
        

public Builder sourceType(OntologyProjectSourceType sourceType) {
    this.sourceType = sourceType;
    return this;
}
            /**
     * Git repository key for git-backed ontology projects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryKey")
private String gitRepositoryKey;

        /**
         * Git repository key for git-backed ontology projects.
         * @param gitRepositoryKey the value to set
         * @return this builder
         **/
        

public Builder gitRepositoryKey(String gitRepositoryKey) {
    this.gitRepositoryKey = gitRepositoryKey;
    return this;
}
            /**
     * Git branch name for git-backed ontology projects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitBranchName")
private String gitBranchName;

        /**
         * Git branch name for git-backed ontology projects.
         * @param gitBranchName the value to set
         * @return this builder
         **/
        

public Builder gitBranchName(String gitBranchName) {
    this.gitBranchName = gitBranchName;
    return this;
}
            /**
     * Workspace-relative Git folder path for git-backed ontology project content.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
private String gitFolderPath;

        /**
         * Workspace-relative Git folder path for git-backed ontology project content.
         * @param gitFolderPath the value to set
         * @return this builder
         **/
        

public Builder gitFolderPath(String gitFolderPath) {
    this.gitFolderPath = gitFolderPath;
    return this;
}


        public CreateOntologyProjectDetails build() {
            CreateOntologyProjectDetails model = new CreateOntologyProjectDetails(this.workspaceId
                , this.key
                , this.displayName
                , this.description
                , this.namespace
                , this.creator
                , this.ontologyVersion
                , this.baseUri
                , this.defaultLanguage
                , this.workspaceBasePath
                , this.sourceType
                , this.gitRepositoryKey
                , this.gitBranchName
                , this.gitFolderPath);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CreateOntologyProjectDetails model) {
                this.workspaceId(model.getWorkspaceId());
    this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.namespace(model.getNamespace());
    this.creator(model.getCreator());
    this.ontologyVersion(model.getOntologyVersion());
    this.baseUri(model.getBaseUri());
    this.defaultLanguage(model.getDefaultLanguage());
    this.workspaceBasePath(model.getWorkspaceBasePath());
    this.sourceType(model.getSourceType());
    this.gitRepositoryKey(model.getGitRepositoryKey());
    this.gitBranchName(model.getGitBranchName());
    this.gitFolderPath(model.getGitFolderPath());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


    
    @com.fasterxml.jackson.annotation.JsonProperty("workspaceId")
    private final String workspaceId;

    
    public String getWorkspaceId() {
        return workspaceId;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

    
    public String getKey() {
        return key;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

    
    public String getDisplayName() {
        return displayName;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

    
    public String getDescription() {
        return description;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("namespace")
    private final String namespace;

    
    public String getNamespace() {
        return namespace;
    }


        /**
     * Creator name written into the generated project.yaml metadata file.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("creator")
    private final String creator;

        /**
     * Creator name written into the generated project.yaml metadata file.
     * @return the value
     **/
    
    public String getCreator() {
        return creator;
    }


        /**
     * Initial semantic ontology version written into project.yaml.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("ontologyVersion")
    private final String ontologyVersion;

        /**
     * Initial semantic ontology version written into project.yaml.
     * @return the value
     **/
    
    public String getOntologyVersion() {
        return ontologyVersion;
    }


        /**
     * Base URI written as baseURI in project.yaml.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("baseUri")
    private final String baseUri;

        /**
     * Base URI written as baseURI in project.yaml.
     * @return the value
     **/
    
    public String getBaseUri() {
        return baseUri;
    }


        /**
     * Default language tag written into project.yaml.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("defaultLanguage")
    private final String defaultLanguage;

        /**
     * Default language tag written into project.yaml.
     * @return the value
     **/
    
    public String getDefaultLanguage() {
        return defaultLanguage;
    }


        /**
     * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("workspaceBasePath")
    private final String workspaceBasePath;

        /**
     * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
     * @return the value
     **/
    
    public String getWorkspaceBasePath() {
        return workspaceBasePath;
    }

    
        /**
     * Project content source. Defaults to VOLUME when omitted.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sourceType")
    private final OntologyProjectSourceType sourceType;

        /**
     * Project content source. Defaults to VOLUME when omitted.
     * @return the value
     **/
    
    public OntologyProjectSourceType getSourceType() {
        return sourceType;
    }


        /**
     * Git repository key for git-backed ontology projects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryKey")
    private final String gitRepositoryKey;

        /**
     * Git repository key for git-backed ontology projects.
     * @return the value
     **/
    
    public String getGitRepositoryKey() {
        return gitRepositoryKey;
    }


        /**
     * Git branch name for git-backed ontology projects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitBranchName")
    private final String gitBranchName;

        /**
     * Git branch name for git-backed ontology projects.
     * @return the value
     **/
    
    public String getGitBranchName() {
        return gitBranchName;
    }


        /**
     * Workspace-relative Git folder path for git-backed ontology project content.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
    private final String gitFolderPath;

        /**
     * Workspace-relative Git folder path for git-backed ontology project content.
     * @return the value
     **/
    
    public String getGitFolderPath() {
        return gitFolderPath;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CreateOntologyProjectDetails(");
        sb.append("workspaceId=").append(String.valueOf(this.workspaceId));
        sb.append(", key=").append(String.valueOf(this.key));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", namespace=").append(String.valueOf(this.namespace));
        sb.append(", creator=").append(String.valueOf(this.creator));
        sb.append(", ontologyVersion=").append(String.valueOf(this.ontologyVersion));
        sb.append(", baseUri=").append(String.valueOf(this.baseUri));
        sb.append(", defaultLanguage=").append(String.valueOf(this.defaultLanguage));
        sb.append(", workspaceBasePath=").append(String.valueOf(this.workspaceBasePath));
        sb.append(", sourceType=").append(String.valueOf(this.sourceType));
        sb.append(", gitRepositoryKey=").append(String.valueOf(this.gitRepositoryKey));
        sb.append(", gitBranchName=").append(String.valueOf(this.gitBranchName));
        sb.append(", gitFolderPath=").append(String.valueOf(this.gitFolderPath));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CreateOntologyProjectDetails)) {
            return false;
        }

        CreateOntologyProjectDetails other = (CreateOntologyProjectDetails) o;
        return java.util.Objects.equals(this.workspaceId, other.workspaceId) &&
            java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.namespace, other.namespace) &&
            java.util.Objects.equals(this.creator, other.creator) &&
            java.util.Objects.equals(this.ontologyVersion, other.ontologyVersion) &&
            java.util.Objects.equals(this.baseUri, other.baseUri) &&
            java.util.Objects.equals(this.defaultLanguage, other.defaultLanguage) &&
            java.util.Objects.equals(this.workspaceBasePath, other.workspaceBasePath) &&
            java.util.Objects.equals(this.sourceType, other.sourceType) &&
            java.util.Objects.equals(this.gitRepositoryKey, other.gitRepositoryKey) &&
            java.util.Objects.equals(this.gitBranchName, other.gitBranchName) &&
            java.util.Objects.equals(this.gitFolderPath, other.gitFolderPath);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.workspaceId == null ? 43 : this.workspaceId.hashCode());
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.namespace == null ? 43 : this.namespace.hashCode());
        result = (result * PRIME) + (this.creator == null ? 43 : this.creator.hashCode());
        result = (result * PRIME) + (this.ontologyVersion == null ? 43 : this.ontologyVersion.hashCode());
        result = (result * PRIME) + (this.baseUri == null ? 43 : this.baseUri.hashCode());
        result = (result * PRIME) + (this.defaultLanguage == null ? 43 : this.defaultLanguage.hashCode());
        result = (result * PRIME) + (this.workspaceBasePath == null ? 43 : this.workspaceBasePath.hashCode());
        result = (result * PRIME) + (this.sourceType == null ? 43 : this.sourceType.hashCode());
        result = (result * PRIME) + (this.gitRepositoryKey == null ? 43 : this.gitRepositoryKey.hashCode());
        result = (result * PRIME) + (this.gitBranchName == null ? 43 : this.gitBranchName.hashCode());
        result = (result * PRIME) + (this.gitFolderPath == null ? 43 : this.gitFolderPath.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/create_ontology_project_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CreateOntologyProjectDetails(object):
    """
    CreateOntologyProjectDetails model.
    """

    #: A constant which can be used with the source_type property of a CreateOntologyProjectDetails.
    #: This constant has a value of "VOLUME"
    SOURCE_TYPE_VOLUME = "VOLUME"

    #: A constant which can be used with the source_type property of a CreateOntologyProjectDetails.
    #: This constant has a value of "GIT"
    SOURCE_TYPE_GIT = "GIT"

    def __init__(self, **kwargs):
        """
        Initializes a new CreateOntologyProjectDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param workspace_id:
            The value to assign to the workspace_id property of this CreateOntologyProjectDetails.
        :type workspace_id: str

        :param key:
            The value to assign to the key property of this CreateOntologyProjectDetails.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this CreateOntologyProjectDetails.
        :type display_name: str

        :param description:
            The value to assign to the description property of this CreateOntologyProjectDetails.
        :type description: str

        :param namespace:
            The value to assign to the namespace property of this CreateOntologyProjectDetails.
        :type namespace: str

        :param creator:
            The value to assign to the creator property of this CreateOntologyProjectDetails.
        :type creator: str

        :param ontology_version:
            The value to assign to the ontology_version property of this CreateOntologyProjectDetails.
        :type ontology_version: str

        :param base_uri:
            The value to assign to the base_uri property of this CreateOntologyProjectDetails.
        :type base_uri: str

        :param default_language:
            The value to assign to the default_language property of this CreateOntologyProjectDetails.
        :type default_language: str

        :param workspace_base_path:
            The value to assign to the workspace_base_path property of this CreateOntologyProjectDetails.
        :type workspace_base_path: str

        :param source_type:
            The value to assign to the source_type property of this CreateOntologyProjectDetails.
            Allowed values for this property are: "VOLUME", "GIT"
        :type source_type: str

        :param git_repository_key:
            The value to assign to the git_repository_key property of this CreateOntologyProjectDetails.
        :type git_repository_key: str

        :param git_branch_name:
            The value to assign to the git_branch_name property of this CreateOntologyProjectDetails.
        :type git_branch_name: str

        :param git_folder_path:
            The value to assign to the git_folder_path property of this CreateOntologyProjectDetails.
        :type git_folder_path: str

        """
        self.swagger_types = {
            'workspace_id': 'str',
            'key': 'str',
            'display_name': 'str',
            'description': 'str',
            'namespace': 'str',
            'creator': 'str',
            'ontology_version': 'str',
            'base_uri': 'str',
            'default_language': 'str',
            'workspace_base_path': 'str',
            'source_type': 'str',
            'git_repository_key': 'str',
            'git_branch_name': 'str',
            'git_folder_path': 'str'
        }

        self.attribute_map = {
            'workspace_id': 'workspaceId',
            'key': 'key',
            'display_name': 'displayName',
            'description': 'description',
            'namespace': 'namespace',
            'creator': 'creator',
            'ontology_version': 'ontologyVersion',
            'base_uri': 'baseUri',
            'default_language': 'defaultLanguage',
            'workspace_base_path': 'workspaceBasePath',
            'source_type': 'sourceType',
            'git_repository_key': 'gitRepositoryKey',
            'git_branch_name': 'gitBranchName',
            'git_folder_path': 'gitFolderPath'
        }

        self._workspace_id = None
        self._key = None
        self._display_name = None
        self._description = None
        self._namespace = None
        self._creator = None
        self._ontology_version = None
        self._base_uri = None
        self._default_language = None
        self._workspace_base_path = None
        self._source_type = None
        self._git_repository_key = None
        self._git_branch_name = None
        self._git_folder_path = None

    @property
    def workspace_id(self):
        """
        Gets the workspace_id of this CreateOntologyProjectDetails.

        :return: The workspace_id of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._workspace_id

    @workspace_id.setter
    def workspace_id(self, workspace_id):
        """
        Sets the workspace_id of this CreateOntologyProjectDetails.

        :param workspace_id: The workspace_id of this CreateOntologyProjectDetails.
        :type: str
        """
        self._workspace_id = workspace_id

    @property
    def key(self):
        """
        **[Required]** Gets the key of this CreateOntologyProjectDetails.

        :return: The key of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this CreateOntologyProjectDetails.

        :param key: The key of this CreateOntologyProjectDetails.
        :type: str
        """
        self._key = key

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this CreateOntologyProjectDetails.

        :return: The display_name of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this CreateOntologyProjectDetails.

        :param display_name: The display_name of this CreateOntologyProjectDetails.
        :type: str
        """
        self._display_name = display_name

    @property
    def description(self):
        """
        Gets the description of this CreateOntologyProjectDetails.

        :return: The description of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this CreateOntologyProjectDetails.

        :param description: The description of this CreateOntologyProjectDetails.
        :type: str
        """
        self._description = description

    @property
    def namespace(self):
        """
        Gets the namespace of this CreateOntologyProjectDetails.

        :return: The namespace of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._namespace

    @namespace.setter
    def namespace(self, namespace):
        """
        Sets the namespace of this CreateOntologyProjectDetails.

        :param namespace: The namespace of this CreateOntologyProjectDetails.
        :type: str
        """
        self._namespace = namespace

    @property
    def creator(self):
        """
        Gets the creator of this CreateOntologyProjectDetails.
        Creator name written into the generated project.yaml metadata file.


        :return: The creator of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._creator

    @creator.setter
    def creator(self, creator):
        """
        Sets the creator of this CreateOntologyProjectDetails.
        Creator name written into the generated project.yaml metadata file.


        :param creator: The creator of this CreateOntologyProjectDetails.
        :type: str
        """
        self._creator = creator

    @property
    def ontology_version(self):
        """
        Gets the ontology_version of this CreateOntologyProjectDetails.
        Initial semantic ontology version written into project.yaml.


        :return: The ontology_version of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._ontology_version

    @ontology_version.setter
    def ontology_version(self, ontology_version):
        """
        Sets the ontology_version of this CreateOntologyProjectDetails.
        Initial semantic ontology version written into project.yaml.


        :param ontology_version: The ontology_version of this CreateOntologyProjectDetails.
        :type: str
        """
        self._ontology_version = ontology_version

    @property
    def base_uri(self):
        """
        Gets the base_uri of this CreateOntologyProjectDetails.
        Base URI written as baseURI in project.yaml.


        :return: The base_uri of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._base_uri

    @base_uri.setter
    def base_uri(self, base_uri):
        """
        Sets the base_uri of this CreateOntologyProjectDetails.
        Base URI written as baseURI in project.yaml.


        :param base_uri: The base_uri of this CreateOntologyProjectDetails.
        :type: str
        """
        self._base_uri = base_uri

    @property
    def default_language(self):
        """
        Gets the default_language of this CreateOntologyProjectDetails.
        Default language tag written into project.yaml.


        :return: The default_language of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._default_language

    @default_language.setter
    def default_language(self, default_language):
        """
        Sets the default_language of this CreateOntologyProjectDetails.
        Default language tag written into project.yaml.


        :param default_language: The default_language of this CreateOntologyProjectDetails.
        :type: str
        """
        self._default_language = default_language

    @property
    def workspace_base_path(self):
        """
        Gets the workspace_base_path of this CreateOntologyProjectDetails.
        Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.


        :return: The workspace_base_path of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._workspace_base_path

    @workspace_base_path.setter
    def workspace_base_path(self, workspace_base_path):
        """
        Sets the workspace_base_path of this CreateOntologyProjectDetails.
        Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.


        :param workspace_base_path: The workspace_base_path of this CreateOntologyProjectDetails.
        :type: str
        """
        self._workspace_base_path = workspace_base_path

    @property
    def source_type(self):
        """
        Gets the source_type of this CreateOntologyProjectDetails.
        Project content source. Defaults to VOLUME when omitted.

        Allowed values for this property are: "VOLUME", "GIT"


        :return: The source_type of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._source_type

    @source_type.setter
    def source_type(self, source_type):
        """
        Sets the source_type of this CreateOntologyProjectDetails.
        Project content source. Defaults to VOLUME when omitted.


        :param source_type: The source_type of this CreateOntologyProjectDetails.
        :type: str
        """
        allowed_values = ["VOLUME", "GIT"]
        if not value_allowed_none_or_none_sentinel(source_type, allowed_values):
            raise ValueError(
                "Invalid value for `source_type`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._source_type = source_type

    @property
    def git_repository_key(self):
        """
        Gets the git_repository_key of this CreateOntologyProjectDetails.
        Git repository key for git-backed ontology projects.


        :return: The git_repository_key of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._git_repository_key

    @git_repository_key.setter
    def git_repository_key(self, git_repository_key):
        """
        Sets the git_repository_key of this CreateOntologyProjectDetails.
        Git repository key for git-backed ontology projects.


        :param git_repository_key: The git_repository_key of this CreateOntologyProjectDetails.
        :type: str
        """
        self._git_repository_key = git_repository_key

    @property
    def git_branch_name(self):
        """
        Gets the git_branch_name of this CreateOntologyProjectDetails.
        Git branch name for git-backed ontology projects.


        :return: The git_branch_name of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._git_branch_name

    @git_branch_name.setter
    def git_branch_name(self, git_branch_name):
        """
        Sets the git_branch_name of this CreateOntologyProjectDetails.
        Git branch name for git-backed ontology projects.


        :param git_branch_name: The git_branch_name of this CreateOntologyProjectDetails.
        :type: str
        """
        self._git_branch_name = git_branch_name

    @property
    def git_folder_path(self):
        """
        Gets the git_folder_path of this CreateOntologyProjectDetails.
        Workspace-relative Git folder path for git-backed ontology project content.


        :return: The git_folder_path of this CreateOntologyProjectDetails.
        :rtype: str
        """
        return self._git_folder_path

    @git_folder_path.setter
    def git_folder_path(self, git_folder_path):
        """
        Sets the git_folder_path of this CreateOntologyProjectDetails.
        Workspace-relative Git folder path for git-backed ontology project content.


        :param git_folder_path: The git_folder_path of this CreateOntologyProjectDetails.
        :type: str
        """
        self._git_folder_path = git_folder_path

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
